R/unused_import_linter.R

```diff
@@ -0,0 +1,64 @@
+#' Check that imported packages are actually used
+#'
+#' @param except_packages Character vector of packages that are ignored.
+#' These are usually attached for their side effects.
+#'
+#' @evalRd rd_tags("unused_import_linter")
+#' @seealso [linters] for a complete list of linters available in lintr.
+#' @export
+unused_import_linter <- function(except_packages = c("bit64", "data.table", "tidyverse")) {
+  Linter(function(source_file) {
+    if (is.null(source_file$full_xml_parsed_content)) return(list())
+
+    import_exprs <- xml2::xml_find_all(
+      source_file$full_xml_parsed_content,
+      paste(
+        "//expr[",
+        "expr[SYMBOL_FUNCTION_CALL[text() = 'library' or text() = 'require']]",
+        "and",
+        "not(SYMBOL_SUB[text() = 'character.only'])",
```

Comment:
this will also skip `library('dplyr', character.only = TRUE)`, right?

Reply:
Yes, it would. Do you suggest adding an `or *[1][self::STR_CONST]` condition to this exception?

Reply:
yes... similar to `missing_package_linter()`